### PR TITLE
Debugger pretty printer files are take into account in test execution time-stamping

### PR DESCRIPTION
**Pull request by k0pernicus**
This PR is proposed to solve the issue #45022.

### CODE REVIEW

File: src/tools/compiletest/src/main.rs [hunk 1]
```diff
@@ -489,15 +489,26 @@ fn stamp(config: &Config, testpaths: &TestPaths) -> PathBuf {
...
-                                         .join("auxiliary")
-                                         .join(aux)));
+        inputs.push(mtime(
+            &testpaths.file.parent().unwrap().join("auxiliary").join(
+                aux,
+            ),
+        ));
+    }
+    // Relevant pretty printer files
+    let pretty_printer_files = [
```
Comment (k0pernicus) [inline src/tools/compiletest/src/main.rs]: For this part, I don't know if the more relevant way to do is to "crawl" the `rust_src_path`, and to check if file which contains in the name "pretty_print". Using this solution, if a member creates a new pretty printer file, this file will be automatically checked in `up_to_date`...
The cons of this automated solution is that we must respect title rules for the pretty printing files.

Comment (michaelwoerister) [reply src/tools/compiletest/src/main.rs]: The explicit list is fine for now.

**Comment by michaelwoerister** [comment]
Looks good to me. You can test it as follows:
* Run `./x.py clean` to bring everything into a predictable state.
* Run `./x.py test --stage 1 src/test/debuginfo`. Most tests should be marked as `ok` (some are always ignored).
* Run `./x.py test --stage 1 src/test/debuginfo` again. Now all tests should be `ignored`.
* Modify one of the python scripts in question.
* Run `./x.py test --stage 1 src/test/debuginfo` again. Now most tests should be `ok` again.

File: src/tools/compiletest/src/main.rs (hunk 1 above)
Comment (michaelwoerister) [reply src/tools/compiletest/src/main.rs]: But could you please add `lldb_batchmode.py` and `lldb_rust_formatters.py` as well?

Comment (k0pernicus) [reply src/tools/compiletest/src/main.rs]: No problem!

**Comment by k0pernicus** [comment]
@michaelwoerister Thanks - I will make those tests after work

**Comment by k0pernicus** [comment]
I just runned the tests, and 98 tests failed... :-(
As an example, I have:
```
test [debuginfo-lldb] debuginfo/associated-types.rs ... FAILED
test [debuginfo-lldb] debuginfo/basic-types-globals-metadata.rs ... FAILED
test [debuginfo-lldb] debuginfo/basic-types-globals.rs ... FAILED
test [debuginfo-lldb] debuginfo/basic-types-metadata.rs ... FAILED
test [debuginfo-lldb] debuginfo/basic-types-mut-globals.rs ... FAILED
test [debuginfo-lldb] debuginfo/basic-types.rs ... FAILED
test [debuginfo-lldb] debuginfo/borrowed-basic.rs ... FAILED
test [debuginfo-lldb] debuginfo/borrowed-c-style-enum.rs ... FAILED
test [debuginfo-lldb] debuginfo/borrowed-enum.rs ... FAILED
test [debuginfo-lldb] debuginfo/borrowed-struct.rs ... FAILED
test [debuginfo-lldb] debuginfo/borrowed-tuple.rs ... FAILED
test [debuginfo-lldb] debuginfo/borrowed-unique-basic.rs ... FAILED
test [debuginfo-lldb] debuginfo/box.rs ... FAILED
test [debuginfo-lldb] debuginfo/boxed-struct.rs ... FAILED
...
```

This is an example of the stacktrace:

```
thread '[debuginfo-lldb] debuginfo/self-in-generic-default-method.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:2617:8

---- [debuginfo-lldb] debuginfo/shadowed-argument.rs stdout ----
	NOTE: compiletest thinks it is using LLDB version 900

error: Error while running LLDB
status: exit code: 1
command: "/usr/bin/python" "/Users/k0pernicus/Code/rust/src/etc/lldb_batchmode.py" "/Users/k0pernicus/Code/rust/build/x86_64-apple-darwin/test/debuginfo/shadowed-argument.stage1-x86_64-apple-darwin" "/Users/k0pernicus/Code/rust/build/x86_64-apple-darwin/test/debuginfo/shadowed-argument.debugger.script"
stdout:
------------------------------------------
LLDB batch-mode script
----------------------
Debugger commands script is '/Users/k0pernicus/Code/rust/build/x86_64-apple-darwin/test/debuginfo/shadowed-argument.debugger.script'.
Target executable is '/Users/k0pernicus/Code/rust/build/x86_64-apple-darwin/test/debuginfo/shadowed-argument.stage1-x86_64-apple-darwin'.
Current working directory is '/Users/k0pernicus/Code/rust'
Creating a target for '/Users/k0pernicus/Code/rust/build/x86_64-apple-darwin/test/debuginfo/shadowed-argument.stage1-x86_64-apple-darwin'
settings set auto-confirm true

------------------------------------------
stderr:
------------------------------------------
Traceback (most recent call last):
  File "/Users/k0pernicus/Code/rust/src/etc/lldb_batchmode.py", line 211, in <module>
    execute_command(command_interpreter, command)
  File "/Users/k0pernicus/Code/rust/src/etc/lldb_batchmode.py", line 84, in execute_command
    print(normalize_whitespace(res.GetOutput()), end='\n')
  File "/Users/k0pernicus/Code/rust/src/etc/lldb_batchmode.py", line 48, in normalize_whitespace
    return re.sub("\s+", " ", s)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/re.py", line 155, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: expected string or buffer
```

Maybe something wrong with my configuration, no?

**Comment by k0pernicus** [comment]
The error is coming from this line of code: `res.GetOutput()`
When debugging, `res.GetOutput()` returns `None`

**Comment by k0pernicus** [comment]
@michaelwoerister The tests to verify the PR are ok.

This is a simple trace:

```
# First batch
...
test result: ok. 98 passed; 0 failed; 10 ignored; 0 measured; 0 filtered out
```

```
# Second batch
test result: ok. 0 passed; 0 failed; 108 ignored; 0 measured; 0 filtered out
```

```
# After modifying a simple line of code in 'gdb_rust_pretty_printing.py'
test result: ok. 98 passed; 0 failed; 10 ignored; 0 measured; 0 filtered out
```

**Comment by k0pernicus** [comment]
@michaelwoerister I made a rebase of my seven commits here: https://github.com/rust-lang/rust/pull/45051/commits/f331433bebbdbc7c15858519e6e8380fd0ea93ba

**Comment by michaelwoerister** [comment]
@k0pernicus GitHub still shows the other commits too, not just the squashed one. 

Following the workflow described at https://github.com/servo/servo/wiki/Beginner%27s-guide-to-rebasing-and-squashing should give you the desired results (which is just a single commit on top of the current `upstream/master`).

**Comment by k0pernicus** [comment]
@michaelwoerister Sorry for that - I forgot the `-p` option when pushing the changes.
It's ok now.

**Comment by michaelwoerister** [comment]
Yes, that looks better! Thank you `:)`

**Pull request merged**